docs/src/udp.rst

```diff
@@ -181,6 +181,23 @@ API
 
     :returns: 0 on success, or an error code < 0 on failure.
 
+.. c:function:: int uv_udp_set_source_membership(uv_udp_t* handle, const char* multicast_addr, const char* interface_addr, const char* source_addr, uv_membership membership)
+
+    Set membership for a source-specific multicast group. Currently implemented only for IPv4.
```

Comment:
There's a good example from `Stevens Unix Network Programming` that supports IPv4 and IPv6 [here](https://github.com/harrifeng/unpbook/blob/master/lib/mcast_join.c#L105-L173) that might help. As @falanger comments it uses `MCAST_JOIN_SOURCE_GROUP` option with the `group_source_req` structure for `IPv6`.